### PR TITLE
Add speed lookup table (still needs real data)

### PR DESCRIPTION
**Pull request by maxh65000**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -74,6 +74,7 @@
   private final Shooter mShooter = Shooter.getInstance();
   private final Intake mIntake = Intake.getInstance();
   private final Storage mStorage = Storage.getInstance();
+  final SpeedLookupTable table = SpeedLookupTable.getInstance();
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java]: This should be a private member variable.
`private final SpeedLookTable mTable = SpeedLookupTable.getInstance();`

File: src/main/java/frc/robot/SpeedLookupTable.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package frc.robot;
+
+import java.util.HashMap;
```
Comment (jhinspeter) [inline src/main/java/frc/robot/SpeedLookupTable.java]: No longer needed?

File: src/main/java/frc/robot/SpeedLookupTable.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+public class SpeedLookupTable {
+  static SpeedLookupTable mInstance;
```
Comment (jhinspeter) [inline src/main/java/frc/robot/SpeedLookupTable.java]: It's either static or it's a member variable. Which is it?

File: src/main/java/frc/robot/SpeedLookupTable.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      mInstance = new SpeedLookupTable();
+    }
+    return mInstance;
+  }
+
+  // the key is distance in meters
+  // the value is velocity in whatever our system uses
+  // note: This MUST be sorted by lowest distance first
+  double[][] map = {
+    {10d, 100d},
```
Comment (jhinspeter) [inline src/main/java/frc/robot/SpeedLookupTable.java:16]: Let's add a comment like:
`/*    {distance in meters, velocity in TBD units} */`
More intuitive, replaces first two line comments above.

**Review by onedeadeye — APPROVED** [review]

File: src/main/java/frc/robot/SpeedLookupTable.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    if (mInstance == null) {
+      mInstance = new SpeedLookupTable();
+    }
+    return mInstance;
+  }
+
+  // the key is distance in meters
+  // the value is velocity in whatever our system uses
+  // note: This MUST be sorted by lowest distance first
+  double[][] map = {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/SpeedLookupTable.java]: This is a member variable. Suggesting a more meaningful name `mInterpolationMap`

File: src/main/java/frc/robot/SpeedLookupTable.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      if(thisDistance == distance){
+        return thisSpeed;
+      }
+
+      lowerBoundDistance = upperBoundDistance;
+      lowerBoundSpeed = upperBoundSpeed;
+      upperBoundDistance = thisDistance;
+      upperBoundSpeed = thisSpeed;
+
+      System.out.println(" lowerBoundDistance: " + lowerBoundDistance + " lowerBoundSpeed: " + lowerBoundSpeed + " upperBoundDistance: " + upperBoundDistance + " upperBoundSpeed: " + upperBoundSpeed);
```
Comment (jhinspeter) [inline src/main/java/frc/robot/SpeedLookupTable.java]: Use Log!

**Review by jhinspeter — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/SpeedLookupTable.java (hunk 4 above)
Comment (maxh65000) [reply src/main/java/frc/robot/SpeedLookupTable.java:16]: Fixed

File: src/main/java/frc/robot/SpeedLookupTable.java (hunk 5 above)
Comment (maxh65000) [reply src/main/java/frc/robot/SpeedLookupTable.java]: Fixed

File: src/main/java/frc/robot/SpeedLookupTable.java (hunk 2 above)
Comment (maxh65000) [reply src/main/java/frc/robot/SpeedLookupTable.java]: Fixed

File: src/main/java/frc/robot/SpeedLookupTable.java (hunk 3 above)
Comment (maxh65000) [reply src/main/java/frc/robot/SpeedLookupTable.java]: fIXED

File: src/main/java/frc/robot/SpeedLookupTable.java (hunk 6 above)
Comment (maxh65000) [reply src/main/java/frc/robot/SpeedLookupTable.java]: Fixed

**Review by jhinspeter — APPROVED** [review]

**Pull request merged**